updates for managed cluster labels (#4656)
open-cluster-management/rhacm-docs: observability/design_grafana.adoc

+
After you complete your updates for the ConfigMap, you can install it to import the dashboard to the Grafana instance.

[#using-managed-cluster-labels]
== Using managed cluster labels in Grafana

> When Observability is enabled in the hub cluster, the `observability-managed-cluster-label-allowlist` ConfigMap is createed in the `open-cluster-management-observability` namespace. The ConfigMap contains a list of managed cluster labels that are maintained by the `observabilty-rbac-query-proxy` pod, to populate a list of label names to filter from within the _ACM - Cluster Overview_ Grafana dashboard. By default, Observability ignores out-of-the-box labels in the `observability-managed-cluster-label-allowlist` ConfigMap.

Reviewer: Created was misspelled.

```suggestion
When Observability is enabled in the hub cluster, the `observability-managed-cluster-label-allowlist` ConfigMap is created in the `open-cluster-management-observability` namespace. The ConfigMap contains a list of managed cluster labels maintained by the `observabilty-rbac-query-proxy` pod, to populate a list of label names to filter from within the _ACM - Cluster Overview_ Grafana dashboard. By default, Observability ignores out-of-the-box labels in the `observability-managed-cluster-label-allowlist` ConfigMap.
```

[source,yaml]
----
data:
  managed_cluster.yaml: |
    ignore_labels:
      - clusterID
      - cluster.open-cluster-management.io/clusterset
      - feature.open-cluster-management.io/addon-application-manager
      - feature.open-cluster-management.io/addon-cert-policy-controller
      - feature.open-cluster-management.io/addon-cluster-proxy
      - feature.open-cluster-management.io/addon-config-policy-controller
      - feature.open-cluster-management.io/addon-governance-policy-framework
      - feature.open-cluster-management.io/addon-iam-policy-controller
      - feature.open-cluster-management.io/addon-observability-controller
      - feature.open-cluster-management.io/addon-search-collector
      - feature.open-cluster-management.io/addon-work-manager
      - installer.name
      - installer.namespace
      - local-cluster
      - name
    labels:
      - cloud
      - vendor
----

> The labels that are enabled are displayed in the drop-down filter on the _ACM - Clusters Overview_ Grafana dashboard. The values are from the `managed_cluster_labels` metric, depending on the `label` key value that is selected.

Reviewer: The metric name is prefixed with `acm`.

```suggestion
The labels that are enabled are displayed in the drop-down filter on the _ACM - Clusters Overview_ Grafana dashboard. The values are from the `acm_managed_cluster_labels` metric, depending on the `label` key value that is selected.
```

      - feature.open-cluster-management.io/addon-work-manager
      - installer.name
      - installer.namespace
      - local-cluster
      - name
    labels:
      - cloud
      - department
      - vendor
----

Enable a managed cluster label that is already disabled by removing the label from the `ignore_labels` list in the `observability-managed-cluster-label-allowlist` ConfigMap.

For example, enable the `local-cluster` and `name` labels, Your `observability-managed-cluster-label-allowlist` ConfigMap might resemble the following content:

[source,yaml]
----
data:
  managed_cluster.yaml: |
    ignore_labels:
      - clusterID
      - installer.name
      - installer.namespace
    labels:
      - cloud
      - vendor
      - local-cluster
      - name
----

The ConfigMap resyncs after 30 seconds to ensure that the cluster labels are updated. After you update the ConfigMap, check the `observability-rbac-query-proxy` pod logs in the `open-cluster-management-observability` namespace to verify where the label is listed. The following information might be displayed in the pod log:

----
enabled managedcluster labels: <label>
----

> //is there a command to run for this verification? 
> From the Grafana dashboard, verify that the label is listed as a valiue in the _Label_ drop-down menu.

Reviewer: Value is misspelled. Also, currently, there is no command to run for this verification. The user can only go to Grafana to view the _Label_ drop-down within the `ACM - Cluster Overview` dashboard. Generating a command could be a future enhancement that we can look into.

```suggestion
//is there a command to run for this verification? 
From the Grafana dashboard, verify that the label is listed as a value in the _Label_ drop-down menu.
```

Author: Updated

[#disabling-managed-cluster-labels]
=== Disabling managed cluster labels

> 

Reviewer: This section is empty for disabling the managed cluster labels.

Author: My bad! I thought i added it. Adding it soon

+
After you complete your updates for the ConfigMap, you can install it to import the dashboard to the Grafana instance.

[#using-managed-cluster-labels]
== Using managed cluster labels in Grafana

> When Observability is enabled in the hub cluster, the `observability-managed-cluster-label-allowlist` ConfigMap is created in the `open-cluster-management-observability` namespace. The ConfigMap contains a list of managed cluster labels maintained by the `observabilty-rbac-query-proxy` pod, to populate a list of label names to filter from within the _ACM - Cluster Overview_ Grafana dashboard. By default, Observability ignores out-of-the-box labels in the `observability-managed-cluster-label-allowlist` ConfigMap.

Reviewer 2: IBM Style recommends not using out of the box as a phrase. Let's try to find a replacement.

![image](https://user-images.githubusercontent.com/108426932/215880806-32d76775-50b3-484f-8586-a0beca8a7fb9.png)

Author: Updated with **custom** to replace out-of-the-box. Thoughts? @dislbenn are there specific tasks that we want to call out?

Reviewer: @dockerymick @oafischer well the ConfigMap is ready for customers to use as is (so in a way it is OOTB), it's just that the ConfigMap will be updated after new labels are detected. Can we replace that statement with something like:

> By default, Observability ignores a subset of labels in the `observability-managed-cluster-label-allowlist` Configmap.

Author: Updated

      - feature.open-cluster-management.io/addon-cert-policy-controller
      - feature.open-cluster-management.io/addon-cluster-proxy
      - feature.open-cluster-management.io/addon-config-policy-controller
      - feature.open-cluster-management.io/addon-governance-policy-framework
      - feature.open-cluster-management.io/addon-iam-policy-controller
      - feature.open-cluster-management.io/addon-observability-controller
      - feature.open-cluster-management.io/addon-search-collector
      - feature.open-cluster-management.io/addon-work-manager
      - installer.name
      - installer.namespace
      - local-cluster
      - name
    labels:
      - cloud
      - vendor
----

The labels that are enabled are displayed in the drop-down filter on the _ACM - Clusters Overview_ Grafana dashboard. The values are from the `acm_managed_cluster_labels` metric, depending on the `label` key value that is selected.

Any label that is listed in the `ignore_labels` keylist of the ConfigMap is removed from the drop-down filter on the _ACM - Clusters Overview_ Grafana dashboard.

[#enabling-adding-managed-cluster-labels]
=== Enabling and adding managed cluster labels

> When you add a new managed cluster label to the `observability-managed-cluster-label-allowlist` ConfigMap, you are enabling the cluster. Add a unique label to the hub cluster, or managed cluster object that is associated with the managed cluster fleet. For example, if you add the label, `department=finance` to a managed cluster, the ConfigMap is updated and might resemble the following changes:

Reviewer: For this line, `"When you add a new managed cluster label to the observability-managed-cluster-label-allowlist ConfigMap, you are enabling the cluster"`, you're not actually enabling the cluster. When the label is added, that label becomes available within Grafana for the user to begin filtering from.

Author: Ahh understood. Thanks for clarifying!

Author: updated

[source,yaml]
----
data:
  managed_cluster.yaml: |
    ignore_labels:
      - clusterID
      - cluster.open-cluster-management.io/clusterset
      - feature.open-cluster-management.io/addon-application-manager
      - feature.open-cluster-management.io/addon-cert-policy-controller
      - feature.open-cluster-management.io/addon-cluster-proxy
      - feature.open-cluster-management.io/addon-config-policy-controller
      - feature.open-cluster-management.io/addon-governance-policy-framework
      - feature.open-cluster-management.io/addon-iam-policy-controller
      - feature.open-cluster-management.io/addon-observability-controller
      - feature.open-cluster-management.io/addon-search-collector
      - feature.open-cluster-management.io/addon-work-manager
      - installer.name
      - installer.namespace
      - local-cluster
      - name
    labels:
      - cloud
      - department
      - vendor
----

Enable a managed cluster label that is already disabled by removing the label from the `ignore_labels` list in the `observability-managed-cluster-label-allowlist` ConfigMap.

> For example, enable the `local-cluster` and `name` labels, Your `observability-managed-cluster-label-allowlist` ConfigMap might resemble the following content:

Reviewer: `Your` is capitalized, this should be lowercase.
```suggestion
For example, enable the `local-cluster` and `name` labels, your `observability-managed-cluster-label-allowlist` ConfigMap might resemble the following content:
```

Author: accepting suggestion

> For example, enable the `local-cluster` and `name` labels, Your `observability-managed-cluster-label-allowlist` ConfigMap might resemble the following content:

Reviewer 2: ```suggestion
For example, enable the `local-cluster` and `name` labels. Your `observability-managed-cluster-label-allowlist` ConfigMap might resemble the following content:
```

Typo (I think)

Author: Updated

----
enabled managedcluster labels: <label>
----

//is there a command to run for this verification? 

From the Grafana dashboard, verify that the label is listed as a value in the _Label_ drop-down menu.

[#disabling-managed-cluster-labels]
=== Disabling managed cluster labels

Exclude a managed cluster label from being listed in the _Label_ drop-down filter. Add the label name to the `ignore_labels` list. For example, your YAML might resemble the following file if you add `local-cluster` and `name` back into the `ignore_labels` list:

[source,yaml]
----
data:
  managed_cluster.yaml: |
    ignore_labels:
      - clusterID
      - installer.name
      - installer.namespace
      - local-cluster
      - name
    labels:
      - cloud
      - vendor
 ----

Check the `observability-rbac-query-proxy` pod logs in the `open-cluster-management-observability` namespace to verify where the label is listed. The following information might be displayed in the pod log:

----
disabled managedcluster label: <label>
----

> 

Reviewer: It seems like the format might've gotten thrown off here.

![Screenshot 2023-01-31 at 3 51 28 PM](https://user-images.githubusercontent.com/28898909/215880414-ea6676b3-ed56-4a8a-9761-47c3b8c3b7a8.png)

Author: Great catch, Updated !

+
After you complete your updates for the ConfigMap, you can install it to import the dashboard to the Grafana instance.

[#using-managed-cluster-labels]
== Using managed cluster labels in Grafana

> When observability is enabled in the hub cluster, the `observability-managed-cluster-label-allowlist` ConfigMap is created in the `open-cluster-management-observability` namespace. The ConfigMap contains a list of managed cluster labels maintained by the `observabilty-rbac-query-proxy` pod, to populate a list of label names to filter from within the _ACM - Cluster Overview_ Grafana dashboard. By default, bservability ignores a subset of labels in the `observability-managed-cluster-label-allowlist` ConfigMap.

Reviewer: Observability is misspelled.

```suggestion
When observability is enabled in the hub cluster, the `observability-managed-cluster-label-allowlist` ConfigMap is created in the `open-cluster-management-observability` namespace. The ConfigMap contains a list of managed cluster labels maintained by the `observabilty-rbac-query-proxy` pod, to populate a list of label names to filter from within the _ACM - Cluster Overview_ Grafana dashboard. By default, observability ignores a subset of labels in the `observability-managed-cluster-label-allowlist` ConfigMap.
```

Author: updated

[#using-managed-cluster-labels]
== Using managed cluster labels in Grafana

> When Observability is enabled in the hub cluster, the `observability-managed-cluster-label-allowlist` ConfigMap is created in the `open-cluster-management-observability` namespace. The ConfigMap contains a list of managed cluster labels maintained by the `observabilty-rbac-query-proxy` pod, to populate a list of label names to filter from within the _ACM - Cluster Overview_ Grafana dashboard. By default, Observability ignores custom labels in the `observability-managed-cluster-label-allowlist` ConfigMap.

Reviewer: I feel like we should replace the word `custom`. I can see some people reading that as by default, any `custom` label that is added, we will ignore.

Author: Ok, what labels specifically get ignored?

Reviewer: The labels that are ignored are the more `uncommon` labels (i.e `local-cluster`). We focused on ignoring labels that might not produce relevant results for users.

Author: Thanks for that info, i missed your [earlier comment](https://github.com/stolostron/rhacm-docs/pull/4656#discussion_r1093377489). Making the update now!

[#using-managed-cluster-labels]
== Using managed cluster labels in Grafana

> When observability is enabled in the hub cluster, the `observability-managed-cluster-label-allowlist` ConfigMap is created in the `open-cluster-management-observability` namespace. The ConfigMap contains a list of managed cluster labels maintained by the `observabilty-rbac-query-proxy` pod, to populate a list of label names to filter from within the _ACM - Cluster Overview_ Grafana dashboard. By default, bservability ignores a subset of labels in the `observability-managed-cluster-label-allowlist` ConfigMap.

Reviewer: Observability is misspelled.

```suggestion
When observability is enabled in the hub cluster, the `observability-managed-cluster-label-allowlist` ConfigMap is created in the `open-cluster-management-observability` namespace. The ConfigMap contains a list of managed cluster labels maintained by the `observabilty-rbac-query-proxy` pod, to populate a list of label names to filter from within the _ACM - Cluster Overview_ Grafana dashboard. By default, observability ignores a subset of labels in the `observability-managed-cluster-label-allowlist` ConfigMap.
```
